### ISSUE TITLE
Variables from tutorial otter-assign notebook not recognized by tests

### ISSUE THREAD

**Issue by connermanuel** [ucbds-infra/otter-grader#182]
**Describe the bug**
Running the instructions from the [tutorial page](https://otter-grader.readthedocs.io/en/latest/tutorial.html) using the provided  [zip file](https://otter-grader.readthedocs.io/en/latest/_static/tutorial.zip) leads to a NameError when `otter assign` runs the tests. Not sure if it has to do with the notebook itself (running through the cells seems fine) or with something in my installation.

**To Reproduce**
1. Download and unpack tutorial.zip
2. cd into the `assign` folder
3. Run command `otter assign assign-demo.ipynb dist` as indicated from tutorial
4. See error: 
```
Line 1, in tests/q3.py 0
Failed example:
    len(data["flavor"].unique()) == 4
Exception raised:
    Traceback (most recent call last):
      File "c:\users\itsco\appdata\local\programs\python\python39\lib\doctest.py", line 1336, in __run
        exec(compile(example.source, filename, "single",
      File "<doctest tests/q3.py 0[0]>", line 1, in <module>
        len(data["flavor"].unique()) == 4
    NameError: name 'data' is not defined
```

**Expected behavior**
`All tests passed!`

**Versions**
Python version: 3.9.1
Otter-Grader version: 1.1.6

**Comment by chrispyles** [comment]
Hmm I've tried following the same steps @connermanuel but am unable to reproduce with Otter 1.1.6:

```
$ otter assign assign-demo.ipynb dist
Generating views...
Generating solutions PDF...
Generating autograder zipfile...
Running tests...
All tests passed!
```

Can you try running assign in debug mode? `otter assign assign-demo.ipynb dist --debug` (this will stop execution of the notebook when an error occurs)

**Comment by connermanuel** [comment]
Seems to be my parser:

```
Traceback (most recent call last):
  File "c:\users\itsco\appdata\local\programs\python\python39\lib\sre_parse.py", line 1039, in parse_template
    this = chr(ESCAPES[this][1])
KeyError: '\\E'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
...
  File "c:\users\itsco\appdata\local\programs\python\python39\lib\re.py", line 210, in sub
    return _compile(pattern, flags).sub(repl, string, count)
  File "c:\users\itsco\appdata\local\programs\python\python39\lib\re.py", line 327, in _subx
    template = _compile_repl(template, pattern)
  File "c:\users\itsco\appdata\local\programs\python\python39\lib\re.py", line 318, in _compile_repl
    return sre_parse.parse_template(repl, pattern)
  File "c:\users\itsco\appdata\local\programs\python\python39\lib\sre_parse.py", line 1042, in parse_template
    raise s.error('bad escape %s' % this, len(this))
re.error: bad escape \E at position 18
```

**Comment by connermanuel** [comment]
Most likely a Git Bash installation issue. Closing this up for now. Thanks and go bears!

**Comment by connermanuel** [comment]
Actually @chrispyles this turned out not to be a git bash issue, may just be a Windows (Win 10) thing in general. Getting the same parsing error with the Windows terminal.

**Comment by chrispyles** [comment]
Yeah, I think it is. I've just merged #197 and released v2.0.6 which removes the use of `re` at all during the execution of notebooks. Can you upgrade to this version and let me know if the error is resolved?

**Comment by connermanuel** [comment]
Works perfectly. Thank you so much to you and to the rest of the infra team!
